apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useContext } from 'react'
+import { useIntl } from 'react-intl'
+
+import {
+  Case,
+  CaseFile,
+  CaseFileCategory,
+  completedCaseStates,
+  isCourtRole,
+} from '@island.is/judicial-system/types'
+import { Box, Text } from '@island.is/island-ui/core'
+
+import PdfButton from '../PdfButton/PdfButton'
+import { caseFiles } from '../../routes/Prosecutor/Indictments/CaseFiles/CaseFiles.strings'
+import { indictmentCaseFilesList as strings } from './IndictmentCaseFilesList.strings'
+import { useFileList } from '../../utils/hooks'
+import { UserContext } from '../UserProvider/UserProvider'
+import SectionHeading from '../SectionHeading/SectionHeading'
+import * as styles from './IndictmentCaseFilesList.css'
+import { courtRecord } from '../../routes/Court/Indictments/CourtRecord/CourtRecord.strings'
+
+interface Props {
+  workingCase: Case
+}
+
+const IndictmentCaseFilesList: React.FC<Props> = (props) => {
+  const { workingCase } = props
+  const { formatMessage } = useIntl()
+  const { onOpen } = useFileList({ caseId: workingCase.id })
+  const { user } = useContext(UserContext)
+
+  const cf = workingCase.caseFiles
+
+  if (!cf || cf.length <= 0) {
+    return null
+  }
+
+  const coverLetters = cf.filter(
+    (file) => file.category === CaseFileCategory.COVER_LETTER,
+  )
+  const indictments = cf.filter(
+    (file) => file.category === CaseFileCategory.INDICTMENT,
+  )
+  const criminalRecords = cf.filter(
+    (file) => file.category === CaseFileCategory.CRIMINAL_RECORD,
+  )
+  const costBreakdowns = cf.filter(
+    (file) => file.category === CaseFileCategory.COST_BREAKDOWN,
+  )
+  const others = cf.filter(
+    (file) =>
+      file.category === CaseFileCategory.CASE_FILE &&
+      !Boolean(file.policeCaseNumber),
+  )
+  const rulings = cf.filter((file) => file.category === CaseFileCategory.RULING)
+  const courtRecords = cf.filter(
+    (file) => file.category === CaseFileCategory.COURT_RECORD,
+  )
+
+  const renderFiles = (caseFiles: CaseFile[]) => {
+    return caseFiles.map((file) => (
+      <Box key={file.id} marginBottom={2} className={styles.caseFileContainer}>
+        <PdfButton
+          caseId={workingCase.id}
+          title={file.name}
+          renderAs="row"
+          handleClick={() => onOpen(file.id)}
+        />
+      </Box>
+    ))
+  }
```

Comment:
Consider moving this to a separate component instead of defining a function inside this component.